rust/src/lib/iface.rs

```diff
@@ -781,7 +781,24 @@ impl MergedInterface {
     pub(crate) fn new(
         desired: Option<Interface>,
         current: Option<Interface>,
+        override_ip: bool,
     ) -> Result<Self, NmstateError> {
+        let mut desired = desired.clone();
+        if override_ip {
+            if let Some(state) = desired.as_mut() {
+                state
+                    .base_iface_mut()
+                    .ipv4
+                    .get_or_insert(Default::default())
+                    .set_explicit_defaults();
+                state
+                    .base_iface_mut()
+                    .ipv6
+                    .get_or_insert(Default::default())
+                    .set_explicit_defaults();
+            }
+        }
```

Comment:
I don't have time to test it now, but it seems to me that we are not actually "overriding". We are still merging with current network state. The actual difference is that we asume some implicit default values in the desired state [1].

Thus, "override" might be a bit missleading. Maybe something like `--ip-implicit-defaults`?

Also, if this option only changes the desired state, we don't need to do this here, we can do it wherever the desired state is created. This would also avoid having to pass the variable `override_ip` over and over.

[1] OK, one of the implicit default value is `enabled: false`, which basically creates the impression of overriding the IP config, but still I don't find it a precise name.

Reply:
Removed. New approach been used. Please check again.